### PR TITLE
inboxer: remove

### PR DESCRIPTION
**Pull request by mgttlinger**
###### Motivation for this change

as discussed in #104183 inboxer shall be removed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4779,7 +4779,7 @@ in
 
   inadyn = callPackage ../tools/networking/inadyn { };
 
-  inboxer = callPackage ../applications/networking/mailreaders/inboxer { };
+  inboxer = throw ''inboxer has been removed as it is no longer maintained and no longer works as Google shut down the inbox service this package wrapped.''
```
Comment (ajs124) [inline pkgs/top-level/all-packages.nix:4782]: Missing `;` and this should go into aliases.nix, not all-packages.nix.

Comment (mgttlinger) [reply pkgs/top-level/all-packages.nix]: fixed

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix]: and please use double quotes as one character.

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -220,6 +220,7 @@ mapAliases ({
   icedtea8_web = adoptopenjdk-icedtea-web; # added 2019-08-21
   icedtea_web = adoptopenjdk-icedtea-web; # added 2019-08-21
   idea = jetbrains; # added 2017-04-03
+  inboxer = throw ''inboxer has been removed as it is no longer maintained and no longer works as Google shut down the inbox service this package wrapped.'';
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:223]:
```suggestion
  inboxer = throw "inboxer has been removed as it is no longer maintained and no longer works as Google shut down the inbox service this package wrapped.";
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (mgttlinger) [reply pkgs/top-level/all-packages.nix]: fixed

**Review by wamserma — APPROVED** [review]

**Pull request merged**